netty/src/main/java/io/grpc/netty/Utils.java

```diff
@@ -168,26 +169,33 @@ private static Http2Headers convertMetadata(Metadata headers) {
   }
 
   public static Status statusFromThrowable(Throwable t) {
+    // First, inspect the causal chain explicitly looking for status exceptions.
     Status s = Status.fromThrowable(t);
     if (s.getCode() != Status.Code.UNKNOWN) {
+      // It was already a status exception, just use it.
       return s;
     }
-    // TODO(ejona): reenable once startup races are resolved; ClosedChannelException is being seen
-    // still. Some tests are asserting UNAVAILABLE and were "working" previously but are now
-    // detecting that our behavior is flaky. See #1330
-    if (false && t instanceof ClosedChannelException) {
-      // ClosedChannelException is used any time the Netty channel is closed. Proper error
-      // processing requires remembering the error that occurred before this one and using it
-      // instead.
-      return Status.UNKNOWN.withCause(t);
-    }
-    if (t instanceof IOException) {
-      return Status.UNAVAILABLE.withCause(t);
-    }
-    if (t instanceof Http2Exception) {
-      return Status.INTERNAL.withCause(t);
+
+    // Assume UNKNOWN unless we find a better status.
+    String msg = t.getMessage();
+    Status status = Status.UNKNOWN.withCause(t).withDescription(msg);
+
+    // Iterate over the causal chain for the error and find the best status.
+    for (Throwable cause : Throwables.getCausalChain(t)) {
+      if (cause instanceof IOException) {
```

Comment:
Please include the ClosedChannelException code. Yes, it is disabled, but there is another bug to try to find the stragglers that prevent making it UNKNOWN instead of UNAVAILABLE (since ClosedChannelException is an IOException).